HB/common/stdpp.elpi

```diff
@@ -78,47 +82,50 @@ bubblesort [X,Y|TL] Rel [Y|Rest1] :- bubblesort [X|TL] Rel Rest1.
 % TODO: pred toposort i:(A -> A -> prop), i:list A, o:list A.
 %       pred edge? i:int, i:int.
 %       toposort edge? [1,2,3,4] TopoList
-pred topovisit i: list (pair A A), i: A,      i: list A, i: list A, o: list A, o: list A.
+func topovisit list (pair A A), A, list A, list A -> list A, list A.
 topovisit _ X VS PS VS PS :- std.mem PS X, !.
 topovisit _ X VS _ _ _ :- std.mem VS X, !, halt "cycle detected.".
 topovisit ES X VS PS VS' [X|PS'] :-
   toporec ES {std.map {std.filter ES (e\ fst e X)} snd} [X|VS] PS VS' PS'.
-pred toporec   i: list (pair A A), i: list A, i: list A, i: list A, o: list A, o: list A.
+func toporec  list (pair A A), list A, list A, list A -> list A, list A.
 toporec _ [] VS PS VS PS.
 toporec ES [X|XS] VS PS VS'' PS'' :-
   topovisit ES X VS PS VS' PS', toporec ES XS VS' PS' VS'' PS''.
-pred toposort i: list (pair A A), i: list A, o: list A.
+func toposort list (pair A A), list A -> list A.
 toposort ES XS XS'' :-
   toporec ES XS [] [] _ XS',
-  std.filter XS' (std.mem XS) XS''.
+  std.filter XS' (std.mem! XS) XS''.
```

Comment:
I think toposort is a function, am I right?
If that’s the case, then we should use std.mem! instead of std.mem to avoid unnecessary 
(and horrible) choice points.